From a10ebae5b09549de6b575989b7bd8a971f765edd Mon Sep 17 00:00:00 2001
From: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
Date: Fri, 15 May 2020 14:13:40 +0530
Subject: [PATCH] hdata: Workaround to hostboot reserved range issue

Workaround until we get below hostboot patch merges to upstream.
  https://rchgit01.rchland.ibm.com/gerrit1/#/c/96121/

With this workaround we will not be able to reserved hostboot load area.
.. which is fine for normal boot. As hostboot is loaded only during IPL
time. But we need above fix for MPIPL to work.

Signed-off-by: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
---
 hdata/memory.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hdata/memory.c b/hdata/memory.c
index a7bbebbf..23c628d5 100755
--- a/hdata/memory.c
+++ b/hdata/memory.c
@@ -795,6 +795,11 @@ static void get_hb_reserved_mem(struct HDIF_common_hdr *ms_vpd)
 		/* remove the HRMOR bypass bit */
 		start_addr &= ~HRMOR_BIT;
 		end_addr &= ~HRMOR_BIT;
+
+		/* Workaround for hostboot bug */
+		if (start_addr == 0)
+			continue;
+
 		if (label_size > HB_RESERVE_MEM_LABEL_SIZE)
 			label_size = HB_RESERVE_MEM_LABEL_SIZE;
 
-- 
2.21.1

